### ISSUE TITLE
NamePatternRule lints CSS props under state-specific style containers (e.g. Multi-State Button `defaultSelectedStyle` / `defaultUnselectedStyle`)

### ISSUE THREAD

**Issue by alex-spyksma** [bw-design-group/ignition-lint#90]
## Summary

`NamePatternRule` incorrectly flags CSS sub-properties of state-specific style containers on the Multi-State Button (and likely other components with similar conventions). Concretely, properties under `defaultSelectedStyle` and `defaultUnselectedStyle` are being linted as if they were user-named properties, even though their children are CSS keys (kebab-case) that the rule already knows to skip for plain `style` / `textStyle` containers.

## Example

![Multi-State Button properties](https://github.com/user-attachments/assets/97ff7c51-791e-47f9-ab56-fa5d303631ec)

For a path like:

```
root.MSB.props.defaultSelectedStyle.background-color
root.MSB.props.defaultUnselectedStyle.font-weight
```

…`NamePatternRule` runs full naming validation on `background-color` / `font-weight`, producing false-positive violations.

## Root cause

`src/ignition_lint/rules/naming/name_pattern.py:365` defines the skip list as plain substrings:

```python
skip_containers = (
    '.style.', '.elementStyle.', '.textStyle.', '.instanceStyle.', '.position.',
    '.props.aspectRatio'
)
```

Because each entry requires a leading `.`, only containers whose key is *exactly* `style` / `textStyle` / etc. are skipped. Anything ending in `Style` (e.g. `defaultSelectedStyle`, `defaultUnselectedStyle`, `disabledStyle`, `indeterminateStyle`, `hoverStyle`, `selectedTextStyle`, `unselectedTextStyle`, …) slips through.

Quick verification:

| Path                                                       | Current behavior |
|------------------------------------------------------------|------------------|
| `root.MSB.props.style.background-color`                    | SKIP (correct)   |
| `root.MSB.props.textStyle.font-family`                     | SKIP (correct)   |
| `root.MSB.props.defaultSelectedStyle.background-color`     | LINT (bug)       |
| `root.MSB.props.defaultUnselectedStyle.font-weight`        | LINT (bug)       |
| `root.MSB.props.disabledStyle.opacity`                     | LINT (bug)       |
| `root.MSB.props.indeterminateStyle.color`                  | LINT (bug)       |
| `root.MSB.props.hoverStyle.background-color`               | LINT (bug)       |

## Proposed fix

Replace the substring list with a regex (or set of compiled regexes) that matches any path segment whose key ends in `Style` (or `TextStyle`), e.g.:

```python
# Matches any container segment ending in 'Style' or 'TextStyle' (with optional [N] array index).
_STYLE_CONTAINER_RE = re.compile(r'\.[A-Za-z]*[Ss]tyle(\[\d+\])?\.')
```

…and apply it alongside the existing `.position.` / `.props.aspectRatio` / SVG `.d` checks in `_should_skip_property`. This keeps the current behavior for `style` / `textStyle` / `elementStyle` / `instanceStyle` and also covers state-prefixed variants without us having to enumerate them.

Open questions for the fix PR:
- Should we keep an explicit allowlist alongside the regex (for clarity / safety), or rely solely on the pattern?
- Are there any Perspective props ending in `Style` whose children are user-named (not CSS) that we'd need to exclude? (None known so far, but worth a sanity check.)

## Acceptance criteria

- A view containing a Multi-State Button with populated `defaultSelectedStyle` / `defaultUnselectedStyle` does not produce naming-rule violations on the CSS keys inside those containers.
- Existing skip behavior for `style`, `textStyle`, `elementStyle`, `instanceStyle` is preserved.
- A golden / unit test covers at least `defaultSelectedStyle`, `defaultUnselectedStyle`, and one of `disabledStyle` / `hoverStyle`.
